Sources/DatadogSDKTesting/Utils/Synced.swift

```diff
@@ -0,0 +1,102 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2020-Present Datadog, Inc.
+ */
+
+import Foundation
+
+final class UnfairLock {
+    private let _lock: os_unfair_lock_t
+    
+    init() {
+        _lock = .allocate(capacity: 1)
+        _lock.initialize(to: .init())
+    }
+    
+    deinit {
+        _lock.deinitialize(count: 1)
+        _lock.deallocate()
+    }
+    
+    func lock() {
+        os_unfair_lock_lock(_lock)
+    }
+    
+    func unlock() {
+        os_unfair_lock_unlock(_lock)
+    }
+    
+    func whileLocked<T>(_ action: () throws -> T) rethrows -> T {
+        os_unfair_lock_lock(_lock)
+        defer { os_unfair_lock_unlock(_lock) }
+        return try action()
+    }
+}
```

Comment:
I think we might be safe 🤷🏻 , but just something to be aware of:

>Unfair locks are low-level locks that block efficiently on contention. They’re useful for protecting code that loads stored resources. However, it’s unsafe to use [os_unfair_lock](https://developer.apple.com/documentation/os/os_unfair_lock) from Swift because it’s a value type and, therefore, doesn’t have a stable memory address. That means when you call [os_unfair_lock_lock](https://developer.apple.com/documentation/os/1646466-os_unfair_lock_lock) or [os_unfair_lock_unlock](https://developer.apple.com/documentation/os/1646463-os_unfair_lock_unlock) and pass a lock object using the & operator, the system may lock or unlock the wrong object.

https://developer.apple.com/documentation/os/osallocatedunfairlock